web_src/js/features/file-view.ts

```diff
@@ -29,26 +29,26 @@ export function initFileView(): void {
       // if no suitable plugin is found, show default view
       if (!success) {
         // show default view raw file link
-        const fallbackText = container.getAttribute('data-fallback-text') || 'View Raw File';
+        const fallbackText = container.getAttribute('data-fallback-text');
 
         container.innerHTML = `
           <div class="view-raw-fallback">
-            <a href="${fileUrl}" class="ui basic button" target="_blank">${fallbackText}</a>
+            <a href="${fileLink}" class="ui basic button" target="_blank">${fallbackText}</a>
```

Comment:
Still XSS

Either, you remember to always escape everything, or, use the helpers like `createElementFromAttrs` and never touch the raw HTML.